template/interface_cpp.njk

```diff
@@ -38,12 +38,15 @@ void {{name}}Bridge::Init(Napi::Env env, Napi::Object exports) {
 
 {{name}}Bridge::{{name}}Bridge(const Napi::CallbackInfo& info)
     : Napi::ObjectWrap<{{name}}Bridge>(info) {
-  // TODO : implement constructor overloading
-  {% for extAttr in extAttrs %} new {{name}} (
-    {% for argument in extAttr.arguments %}
-      NativeTypeTraits<IDL{{argument.type | camelcase-}}>::NativeValue(info.Env(), info[{{loop.index - 1}}]){% if not loop.last %},{% endif %}
-    {% endfor %})
-  {% endfor %};
+  // TODO : for implement method overloading we should be check argument type also.
+  {% for extAttr in extAttrs %} 
+    if (info.Length() == {{extAttr.arguments.length}}) {
+      new {{name}} (
+      {% for argument in extAttr.arguments %} 
+        NativeTypeTraits<IDL{{argument.type | camelcase-}}>::NativeValue(info.Env(), info[{{loop.index - 1}}]){% if not loop.last %},{% endif %}
+      {% endfor %});
```

Comment:
I think we need ```return;``` here.

Reply:
If we add ```return;``` here, some logic error is occurred when there are constructors with same number of arguments.

Actually, if we do not add ```return;``` also, we got some javascript type error. 

In other words, both are errors,
So I should implement the type checker as quickly as possible. ;(

By the way I have a question.
@romandev Do you think that it is okay to use condition based on info like this?

Reply:
@yjaeseok Thank you for reply,
I'm sorry but I didn't understand your comment fully.
Even if some constructors has the same number of arguments, the constructor is only called once, right?
You already checked length of arguments and anyway some constructor will be called even if there are two constructors with same number of arguments. So, we can anyway return after calling the constructor.

For example, there are two constructors.
- Constructor(int a, int b);
- Constructor(int a, int b, int c);

Generator will be the following codes.
```c++
...
if (info.Length() == 2) {
  new Constructor(...);
  return;    // We need return; here. If not so, we will check info.Length() again unnecessarily.
}
if (info.Length() == 3) {
  new Constructor(...);
}
...
```

Am I missing something? Please correct me :)

Reply:
> By the way I have a question.
> @romandev Do you think that it is okay to use condition based on info like this?

looks good to me; We can not sure which constructor is called by JS developer in code generating time. There is no way to know the information. So, your approach is best I think.

Reply:
Okay, I would like to introduce you some exception case.
for example there are two constructors with same number of arguments.

in calculator.h
```c++
Calculator(long arg1, long arg2);
Calculator(string msg1, string msg2);
```
and calculator.idl
```
[
  Calculator(long arg1, long arg2),
  Calculator(string msg1, string msg2)
]
```

For this example, the calculator_bridge.cc will probably be created like this:

```c++
 59     if (info.Length() == 2) {
 60       new Calculator (
 61 
 62         NativeTypeTraits<IDLLong>::NativeValue(info.Env(), info[0]),
 63 
 64         NativeTypeTraits<IDLLong>::NativeValue(info.Env(), info[1])
 65       );
 66       return;
 67     }
 68 
 69     if (info.Length() == 2) {
 70       new Calculator (
 71         
 72         NativeTypeTraits<IDLString>::NativeValue(info.Env(), info[0]),
 73         
 74         NativeTypeTraits<IDLString>::NativeValue(info.Env(), info[1])
 75       );
 76       return;
 77     }
```

In this case, user cannot call ```Calculator(string msg1, string msg2); ```. Because it is returned on line 66.

If we do not return also, javascript error is occurred on line 62.

Reply:
If you have any question? please tell me back!